EIPS/eip-7950.md

```diff
@@ -0,0 +1,50 @@
+---
+eip: 7950
+title: Encode chain id with transaction hash
+description: Way to encode a chain ID and transaction hash into a unique string format
+author: Lauri Peltonen (@microbecode)
+discussions-to: https://ethereum-magicians.org/t/a-new-standard-for-encoding-chain-id-transaction-hash/23782
+status: Draft
+type: Standards Track
+category: Interface
+created: 2025-05-22
+requires: 155
+---
+
+## Abstract
+
+This standard proposes a way to encode the combination of a chain ID and a transaction hash into one string.
+
+## Motivation
+
+Looking up a transaction by its hash always requires the context of the chain - a transaction hash alone is not enough to identify the used chain. If the chain information is included in the string itself, finding the right chain for the transaction is easy.
```

Comment:
I would include here that for non-legacy type transactions the `chainId` is included in the transaction object, which thus uniquely maps the data to a chain-id unique hash. The same applies to EIP-155 replay protected transactions. But, obviously, this map works only from data => hash and not the other way around.

(This caused slight confusion in my head on why to include chainId "in" the hash (since this is "part" of it), but this is not the case, it adds the chain id as identifer to the hash to also make the "modified hash" => chain id map possible (side-by-side the other way around))

Reply:
Thanks for the feedback!

Good point about the transaction object. A lot of places may include the chain id, but nobody really wants to store a whole transaction object in places just to keep the chain id reference.

I added now a mention. Thanks again!